src/scripts/battleFrontier/BattleFrontierMilestones.ts

```diff
@@ -113,6 +113,10 @@ BattleFrontierMilestones.addMilestone(new BattleFrontierMilestoneItem(550, 'Elec
 BattleFrontierMilestones.addMilestone(new BattleFrontierMilestoneItem(575, 'Protector', 15, new MaxRegionRequirement(GameConstants.Region.sinnoh)));
 BattleFrontierMilestones.addMilestone(new BattleFrontierMilestoneItem(600, 'Reaper_cloth', 15, new MaxRegionRequirement(GameConstants.Region.sinnoh)));
 BattleFrontierMilestones.addMilestone(new BattleFrontierMilestoneItem(660, 'Sachet', 15, new MaxRegionRequirement(GameConstants.Region.kalos)));
+BattleFrontierMilestones.addMilestone(new BattleFrontierMilestonePokemon(666, 'Vivillon (Poké Ball)', new MultiRequirement([
+    new QuestLineStepCompletedRequirement('The Great Vivillon Hunt!', 34),
+    new QuestLineCompletedRequirement('The Great Vivillon Hunt!', GameConstants.AchievementOption.less),
```

Comment:
I don't think we need the `less` requirement, as the milestone goes away after you achieve it the first time?

Reply:
That's true going forward. Because of how it was added and removed to the reward list the milestone wasn't being tracked in the save, so players who have already completed the quest and obtained it will see it re-appear in the reward list.

I suppose adding it to the save in the update function and getting rid of that extra requirement would be better then.

Reply:
Updated to remove the extra requirement and added an update function that will add it to the obtained milestones if the quest is completed.